cli/azd/pkg/devcenter/manager.go

```diff
@@ -342,3 +348,63 @@ func (m *manager) Outputs(
 
 	return outputs, nil
 }
+
+func mapDevCenterTypeToParamType(devCenterType devcentersdk.OutputParameterType) (provisioning.ParameterType, error) {
+	switch strings.ToLower(string(devCenterType)) {
+	case string(devcentersdk.OutputParameterTypeString):
+		return provisioning.ParameterTypeString, nil
+	case string(devcentersdk.OutputParameterTypeBoolean):
+		return provisioning.ParameterTypeBoolean, nil
+	case string(devcentersdk.OutputParameterTypeNumber):
+		return provisioning.ParameterTypeNumber, nil
+	case string(devcentersdk.OutputParameterTypeObject):
+		return provisioning.ParameterTypeObject, nil
+	case string(devcentersdk.OutputParameterTypeArray):
+		return provisioning.ParameterTypeArray, nil
+	default:
+		return "", fmt.Errorf("unexpected output parameter type: '%s'", string(devCenterType))
+	}
+}
+
+// Creates a normalized view of the azure output parameters and resolves inconsistencies in the output parameter name
+// casings.
+func createOutputParameters(
+	outputsResponse *devcentersdk.OutputListResponse,
+) (map[string]provisioning.OutputParameter, error) {
+	outputParams := map[string]provisioning.OutputParameter{}
+
+	for key, devCenterParam := range outputsResponse.Outputs {
+		// To support BYOI (bring your own infrastructure) scenarios we will default to UPPER when canonical casing
+		// is not found in the parameters file to workaround strange azure behavior with OUTPUT values that look
+		// like `azurE_RESOURCE_GROUP`
+		paramName := strings.ToUpper(key)
+		value := devCenterParam.Value
+
+		paramType, err := mapDevCenterTypeToParamType(devCenterParam.Type)
+		if err != nil {
+			return nil, err
+		}
+
+		requiresJsonMarshalling := slices.Contains(
+			[]devcentersdk.OutputParameterType{
+				devcentersdk.OutputParameterTypeObject,
+				devcentersdk.OutputParameterTypeArray,
+			}, devCenterParam.Type)
+
+		if requiresJsonMarshalling {
+			jsonBytes, err := json.Marshal(devCenterParam.Value)
```

Comment:
Just checking on this -- what requires us to marshal complex types? It's interesting that we'd allow for `int` and `bool` to be set directly, but we'd represent the complex types as JSON-string. 

I know in [provisioning manager](https://github.com/Azure/azure-dev/blob/e03b5fccc87cceafce9e83f16688b3c7a7bb8a97/cli/azd/pkg/infra/provisioning/manager.go#L156), we'd end up marshaling it for environment outputting, but the Bicep provider doesn't serialize the output either, it just passes the complex types through.

Reply:
I believe that ARM deployments API already returns complex types as JSON strings and this is just matching that expectation since these outputs are being set in `.env` files.

Reply:
Indeed the API is returning a JSON string.

When we call `json.Unmarshal` however, the value would be deserialized as a weakly-typed `map[string]interface{}`. I'd like to keep a consistent model of what `Outputs` should represent -- it's the deserialized JSON (weakly-typed). Serializing it here as a JSON string makes it harder to reason about. 

Keeping it as `map[string]interface{}` instead of `string` also make ADE's provision provider consistent with other providers. Here is what happens today in Bicep provisioning provider when it returns the `Outputs` to the provision manager, it's a `map[string]interface{}`:
![image](https://github.com/Azure/azure-dev/assets/2322434/7672fb16-23f2-4150-8055-ab53cf052574)

We only serialize all values to their string-representation when we're saving it to the `.env` file in provision.Manager:
![image](https://github.com/Azure/azure-dev/assets/2322434/fc322999-1354-40d8-ba16-2c4de94f927c)

My best guess is that if we removed JSON marshaling here, the observable-behavior of the system would work just the same, we just made the internal representation more consistent.

Reply:
Ok, so you are saying that our `UpdateEnvironment` func already handles the complexities of marshalling complex types into string... If that is the case then I agree that the Outputs should be consistent with this experience.